### ISSUE TITLE
clarify spot instance pricing

### ISSUE THREAD

**Issue by casperdcl** [iterative/cml.dev#285]
1. a high enough max bidding price (confusingly named `--cloud-spot-price`) is required to get a spot instance
2. after a spot instance is running:
    - [if the spot price rises above the maximum (specified above), AWS EC2 interrupts your spot instance](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html)
    - also, AWS can (and does) often interrupt it anyway (regardless of your bidding price) unpredictably
So basically there's no way to make spot instances never interrupt. Instead you need to either use on-demand (costs $$$ more ongoing), or program some [recovery logic](https://iterativeai.slack.com/archives/C01900GSB4J/p1657801783794509?thread_ts=1657733460.262019&cid=C01900GSB4J) (costs time/dev effort one-time)
- what about other cloud providers?
- what are the conditions for resuming? what does the user need to do (#207)?

**Comment by jorgeorpinel** [comment]
Would this go into an existing doc somewhere?
